docs/reference/environment-variables.rst

```diff
@@ -8,86 +8,74 @@
 Environment variables
 *********************
 
+.. envvar:: ARCHFLAGS
 
-.. todo::
+   This environmental variable is used for supporting architecture cross
+   compilation on macOS in a way compatible with setuptools_. It is ignored on
+   all other platforms. It can be set to ``-arch arm64`` or to ``-arch
+   x86_64`` for compiling for the aarch64 and the x86-64 architectures
+   respectively. Setting this environment variable to any other value is not
+   supported.
 
-   - Document ``ARCHFLAGS`` (see `#226`_)
+   The macOS toolchain allows architecture cross compilation passing the
+   ``-arch`` flat to the compilers. ``meson-python`` inspects the content of
+   this environment variable and synthesizes a Meson `cross build definition
+   file`_ with the appropriate content, and passes it to ``meson setup`` via
+   the ``--cross-file`` option.
 
+   Support for this environment variable is maintained only for compatibility
+   with existing tools and is not the recommended solution for cross
```

Comment:
consider inserting, here or higher up where setuptools is mentioned: "`cibuildwheel` in particular" (I _think_ that's the only real reason)